server/src/internalClusterTest/java/org/elasticsearch/indices/DataStreamIT.java

```diff
@@ -210,6 +219,135 @@ public void testOtherWriteOps() throws Exception {
         }
     }
 
+    public void testTimeStampValidation() throws Exception {
+        // Adding a template with a mapping without a timestamp field and expect data stream creation to fail.
+        {
+            PutComposableIndexTemplateAction.Request createTemplateRequest = new PutComposableIndexTemplateAction.Request("logs-foo");
+            createTemplateRequest.indexTemplate(
+                new ComposableIndexTemplate(
+                    List.of("logs-*"),
+                    new Template(null, new CompressedXContent("{}"), null),
+                    null, null, null, null,
+                    new ComposableIndexTemplate.DataStreamTemplate("@timestamp"))
+            );
+            client().execute(PutComposableIndexTemplateAction.INSTANCE, createTemplateRequest).actionGet();
+            Exception e = expectThrows(IllegalArgumentException.class,
+                () -> client().index(new IndexRequest("logs-foobar").opType(CREATE).source("{}", XContentType.JSON)).actionGet());
+            assertThat(e.getMessage(), equalTo("expected timestamp field [@timestamp], " +
+                "but found no timestamp field for backing index [logs-foobar-000001]"));
+        }
+        // Adding a template with a mapping with a timestamp field and expect data stream creation to succeed.
```

Comment:
I think these should each be separate test cases, instead of squeezing N into a single test. It makes for smaller more digestible tests, with clear names, whose failures are easier to understand.

Reply:
Thanks, will do that in the followup prs.